tests/vxlan/vnet_utils.py

```diff
@@ -0,0 +1,122 @@
+                                        num_vnet=num_vnet, num_routes=num_routes, num_endpoints=num_endpoints))
+
+def render_template_to_host(template_name, host, dest_file, *template_args, **template_kwargs):
+    """
+    @summary: Renders a template with the given arguments and copies it to the host
+    @param template_name: A template inside the "templates" folder (without the preceding "templates/")
+    @param host: The host device to copy the rendered template to
+    @param dest_file: The location on the host to copy the rendered template to
+    @param *template_args: Any arguments to be passed to j2 during rendering
+    @param **template_kwargs: Any keyword arguments to be passed to j2 during rendering
+    """
+    # Combine all dictionaries given in template_args
+    template_args_iter = iter(template_args)
+    try:
+        combined_template_args = next(template_args_iter).copy()
+        for arg in template_args_iter:
+            combined_template_args.update(arg)
+    except StopIteration:
+        combined_template_args = {}
+
+    rendered = Template(open(path.join("templates",template_name)).read()) \
+                        .render(combined_template_args, **template_kwargs)
+
+    host.copy(content=rendered, dest=dest_file)
+
+def generate_dut_config_files(duthost, mg_facts):
+    """
+    @summary: Generate VNET and VXLAN config files and copy them to DUT.
+    @param duthost: DUT host object
+    @param mg_facts: Minigraph facts
+    """
+    logger.info("Generating config files and copying to DUT")
+
+    vnet_switch_config = [{
+        "SWITCH_TABLE:switch": {
+            "vxlan_port": vc.VXLAN_PORT,
+            "vxlan_router_mac": vc.VXLAN_MAC
+        },
+        "OP": "SET"
+    }]
+
+    duthost.copy(content=json.dumps(vnet_switch_config, indent=4), dest=vc.DUT_VNET_SWITCH_CONFIG)
+
+
+    render_template_to_host("vnet_vxlan.j2", duthost, vc.DUT_VNET_CONF, vc.VNET_CONFIG, mg_facts, ipv6_vxlan_test=vc.IPV6_VXLAN_TEST)
+    render_template_to_host("vnet_interface.j2", duthost, vc.DUT_VNET_INTF_CONFIG, vc.VNET_CONFIG)
+    render_template_to_host("vnet_nbr.j2", duthost, vc.DUT_VNET_NBR_JSON, vc.VNET_CONFIG)
+    render_template_to_host("vnet_routes.j2", duthost, vc.DUT_VNET_ROUTE_CONFIG, vc.VNET_CONFIG, op="SET")
+
+def apply_dut_config_files(duthost):
+    """
+    @summary: Applies config files on disk
+    @param duthost: DUT host object
+    """
+    logger.info("Applying config files on DUT")
+
+    config_files = ["/tmp/vnet.intf.json", "/tmp/vnet.nbr.json"]
+    if vc.APPLY_NEW_CONFIG:
+        config_files.append("/tmp/vnet.conf.json")
+
+    for config in config_files:
+        duthost.shell("sonic-cfggen -j {} --write-to-db".format(config))
+        sleep(3)
+
+    duthost.shell("docker cp {} swss:/vnet.route.json".format(vc.DUT_VNET_ROUTE_CONFIG))
+    duthost.shell("docker cp {} swss:/vnet.switch.json".format(vc.DUT_VNET_SWITCH_CONFIG))
```

Comment:
Is this still needed if `APPLY_NEW_CONFIG` isn't true?

Reply:
This is behavior taken from the Ansible test. @prsunny Is there ever a situation where we want to place the config files on the DUT without actually applying them? The `APPLY_NEW_CONFIG` var is meant to replace `config` which is set [here in the Ansible test](https://github.com/Azure/sonic-mgmt/blob/54264e2d97185235ba823aea04827abee9782c3b/ansible/roles/test/tasks/vnet_vxlan.yml#L87)

Reply:
Discussed offline, this entire method can be bypassed if the `--skip_apply_config` CLI parameter is passed. Fixed in next push